fastlane/lib/fastlane/actions/oclint.rb

```diff
@@ -23,7 +23,11 @@ def self.run(params)
         select_regex = params[:select_regex] if params[:select_regex] # Overwrite deprecated select_reqex
         exclude_regex = params[:exclude_regex]
 
-        files = JSON.parse(File.read(compile_commands)).map { |compile_command| compile_command['file'] }
+        files = JSON.parse(File.read(compile_commands)).map do |compile_command|
+          file = compile_command['file']
+          File.exist?(file) ? file : compile_command['directory'] + '/' + file
```

Comment:
good suggestion for File.join(...) (we are not ruby programmers :P, we patch some actions to fastlane works)

and yes, the path is correct.
Our workspace directory structure is

```
App_App_ios
|-app
|-framework
|-staticlib
|--StaticLibShared
   |--StaticLibShared
      |--ClassStaticLibShared.h
      |--ClassStaticLibShared.m
      |--StaticLibShared.h
      |--StaticLibShared.m
   |--StaticLibShared.xcodeproj
   |--ClassStaticLibShared.plist
   |--StaticLibShared.plist
|--StaticLibUsingFramework
workspace.xcworkspace

```